src/preamble.js

```diff
@@ -1164,6 +1164,16 @@ assert(typeof Int32Array !== 'undefined' && typeof Float64Array !== 'undefined'
 #endif
 
 #if IN_TEST_HARNESS
+
+// Test runs in browsers should always be free from uncaught exceptions. If an uncaught exception is thrown, we fail browser test execution in the REPORT_RESULT() macro to output an error value.
+if (typeof window !== 'undefined') {
+  window.addEventListener('error', function(e) {
+    if (e == 'SimulateInfiniteLoop' || e.message.indexOf('SimulateInfiniteLoop') != -1) return;
```

Comment:
> We have a place in postamble.js that just checks the string. Is that incorrect, then?

According to MDN, it looks like `e` can be an event when `window.addEventListener` is used, and a string message if `window.onerror` is used. When I run the test locally, I don't get a string in that added code, but an event.

> I don't even see that line of code (that checks for SimulateInfiniteLoop in two ways) is even reached in browser.test_sdl_touch..?

That code is not supposed to be reached in test `browser.test_sdl_touch` because the test does not throw `"SimulateInfiniteLoop"`. If you run `browser.test_emscripten_main_loop` without the `" || e.message.indexOf('SimulateInfiniteLoop') != -1"` part, then that test fails to find that thrown exception and does not pass.